docs/en/observability/logs-stream.asciidoc

```diff
@@ -143,10 +143,220 @@ If you're not seeing your log files in {kib}, check the following in the `elasti
+
+Optionally, you can set the following values for the `date` processor:
+
+- `timezone` – Sets the timezone to use when parsing the date.
+- `target_field` – Sets the output field for the timestamp.
+- `output_format` – Changes the format when writing the date to the timestamp field. Must be a valid java time pattern.
+
+[discrete]
+[[logs-stream-index-template]]
+==== Configure your data stream with an index template
+
+After creating your ingest pipeline, you need to index your log data to a data stream using an index template. 
+
+The following command creates an index template:
+
+[source,console]
+----
+PUT _index_template/logs-generic-default-template
+{
+  "index_patterns": [ "logs-generic-default" ],
+  "data_stream": { },
+  "priority": 500,
+  "template": {
+    "settings": {
+      "index.default_pipeline":"logs-generic-default"
+    }
+  }
+}
+----
+
+Set the following values for the index template:
+
+- `index_patterns`– The index pattern needs to match your log data stream. Naming conventions for data streams are `<type>-<dataset>-<namespace>`. If you're using default values, your logs data stream is named `logs-generic-default`. Data that matches this pattern will go through your pipeline.
+- `data_stream` – Enables data streams.
+- `priority` – Index templates with higher priority take precedence over lower priority. If a data stream matches multiple index templates, the template with the higher priority is used. Built-in templates have a priority of `200`, so we recommend a priority higher than that.
+- `index.default_pipeline` – The name of the ingest pipeline you created.
+
+[discrete]
+[[logs-stream-create-data-stream]]
+==== Create your data stream
+
+Create your data stream using the {fleet-ref}/data-streams.html#data-streams-naming-scheme[data stream naming scheme]. The name needs to match the name of your pipeline. For this example, we'll use the default log data stream name, `logs-generic-default`, and the example log:
+
+[source,console]
+----
+POST logs-generic-default/_doc
+{
+  "message": "2023-08-08T13:45:12.123Z WARN 192.168.1.101 Disk usage exceeds 90%."
+}
+----
+
+Now look at the details using the following command:
+
+[source,JSON]
+----
+GET /<index-name>/_doc/<document-id>
+----
+
+Under `_source`, you can see the `timestamp` field that was extracted by the dissect processor and the `@timestamp` field that was extracted by the date processor:
+
+[source]
+----
+{
+  "_index": ".ds-logs-generic-default-2023.08.09-000001",
+  "_id": "StG93IkByBSsmSqxzBqO",
+  "_version": 1,
+  "_seq_no": 3,
+  "_primary_term": 1,
+  "found": true,
+  "_source": {
+    "message": "WARN 192.168.1.101 Disk usage exceeds 90%.",
+    "timestamp": "2023-08-08T13:45:12.123Z",
+    "@timestamp": "2023-08-08T13:45:12.123Z"
+  }
+}
+----
+
+You can now use the `@timestamp` field to sort your logs by the date and time they happened.
+
+[discrete]
+[[logs-stream-timestamp-troubleshooting]]
+==== Troubleshooting
```

Comment:
@ruflin do you have any suggestions for issues users might run into?

Reply:
invalid timestamps, wrong time zone, figuring out the format. You have listed some of the config options above. But debugging it is painful.

@felixbarny @graphaelli I always found it very painful debugging "non standard" timestamp formats and trying to get things work. I wonder if we could improve this in the future too.